src/battle_anim_throw.c

```diff
@@ -943,7 +941,7 @@ void AnimTask_IsBallBlockedByTrainer(u8 taskId)
     DestroyAnimVisualTask(taskId);
 }
 
-u8 ItemIdToBallId(u16 ballItem)
+enum PokeBall ItemIdToBallId(u16 ballItem)
```

Comment:
I understand you may want to consider this out of scope, but imo we should really remove this function and define the `BALL_` ids in `items.h` in some field. This unification would be more in line with our current expansion approach.

Reply:
You can't fully remove it because of the fail safe but I'm fine with this change

Reply:
Then perhaps this function can just be "evaluate the field and provide the failsafe in case it's `NULL`"?

Reply:
I could just keep the function and check the struct id instead. That would also reduce the number of changes I would need to make.